core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -272,17 +272,17 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
 
     // Disable the background thread during tests.
     if (!conf.contains(IS_TESTING)) {
-      // A task that periodically checks for event log updates on disk.
-      logDebug(s"Scheduling update thread every $UPDATE_INTERVAL_S seconds")
-      pool.scheduleWithFixedDelay(
-        getRunner(() => checkForLogs()), 0, UPDATE_INTERVAL_S, TimeUnit.SECONDS)
-
       if (conf.get(CLEANER_ENABLED)) {
         // A task that periodically cleans event logs on disk.
         pool.scheduleWithFixedDelay(
           getRunner(() => cleanLogs()), 0, CLEAN_INTERVAL_S, TimeUnit.SECONDS)
       }
 
+      // A task that periodically checks for event log updates on disk.
```

Comment:
We can only control this interleaving to a certain extent. For example, when `UPDATE_INTERVAL_S` is configured to be less than `CLEAN_INTERVAL_S` (which is usually the case), we are still going to be performing the `checkForLogs` before we `cleanLogs`.

Reply:
> We can only control this interleaving to a certain extent. For example, when `UPDATE_INTERVAL_S` is configured to be less than `CLEAN_INTERVAL_S` (which is usually the case), we are still going to be performing the `checkForLogs` before we `cleanLogs`.

I understand it, thanks